patch-package
--- a/node_modules/react-infinite-scroller/dist/InfiniteScroll.js
+++ b/node_modules/react-infinite-scroller/dist/InfiniteScroll.js
@@ -108,6 +108,9 @@ var InfiniteScroll = function (_Component) {
       if (this.props.useWindow === false) {
         scrollEl = this.scrollComponent.parentNode;
       }
+      if (this.props.getScrollElement != null) {
+        scrollEl = this.props.getScrollElement().parentNode;
+      }
 
       scrollEl.removeEventListener('mousewheel', this.mousewheelListener, this.options ? this.options : this.props.useCapture);
     }
@@ -118,6 +121,9 @@ var InfiniteScroll = function (_Component) {
       if (this.props.useWindow === false) {
         scrollEl = this.getParentElement(this.scrollComponent);
       }
+      if (this.props.getScrollElement != null) {
+        scrollEl = this.getParentElement(this.props.getScrollElement());
+      }
 
       scrollEl.removeEventListener('scroll', this.scrollListener, this.options ? this.options : this.props.useCapture);
       scrollEl.removeEventListener('resize', this.scrollListener, this.options ? this.options : this.props.useCapture);
@@ -150,6 +156,10 @@ var InfiniteScroll = function (_Component) {
         scrollEl = parentElement;
       }
 
+      if (this.props.getScrollElement != null) {
+        scrollEl = this.getParentElement(this.props.getScrollElement());
+      }
+
       scrollEl.addEventListener('mousewheel', this.mousewheelListener, this.options ? this.options : this.props.useCapture);
       scrollEl.addEventListener('scroll', this.scrollListener, this.options ? this.options : this.props.useCapture);
       scrollEl.addEventListener('resize', this.scrollListener, this.options ? this.options : this.props.useCapture);
@@ -238,7 +248,8 @@ var InfiniteScroll = function (_Component) {
           useCapture = renderProps.useCapture,
           useWindow = renderProps.useWindow,
           getScrollParent = renderProps.getScrollParent,
-          props = _objectWithoutProperties(renderProps, ['children', 'element', 'hasMore', 'initialLoad', 'isReverse', 'loader', 'loadMore', 'pageStart', 'ref', 'threshold', 'useCapture', 'useWindow', 'getScrollParent']);
+          getScrollElement = renderProps.getScrollElement,
+          props = _objectWithoutProperties(renderProps, ['children', 'element', 'hasMore', 'initialLoad', 'isReverse', 'loader', 'loadMore', 'pageStart', 'ref', 'threshold', 'useCapture', 'useWindow', 'getScrollParent', 'getScrollElement']);
 
       props.ref = function (node) {
         _this2.scrollComponent = node;
@@ -275,7 +286,8 @@ InfiniteScroll.propTypes = {
   getScrollParent: _propTypes2.default.func,
   threshold: _propTypes2.default.number,
   useCapture: _propTypes2.default.bool,
-  useWindow: _propTypes2.default.bool
+  useWindow: _propTypes2.default.bool,
+  getScrollElement: _propTypes2.default.func
 };
 InfiniteScroll.defaultProps = {
   element: 'div',
@@ -288,7 +300,8 @@ InfiniteScroll.defaultProps = {
   isReverse: false,
   useCapture: false,
   loader: null,
-  getScrollParent: null
+  getScrollParent: null,
+  getScrollElement: null
 };
 exports.default = InfiniteScroll;
 module.exports = exports['default'];
